tsl/src/compression/compression.c

```diff
@@ -2766,6 +2766,9 @@ fill_predicate_context(Chunk *ch, List *predicates, List **filters, List **index
 				else
 					continue;
 
+				/* ignore system-defined attributes */
+				if (var->varattno < 0)
```

Comment:
I am wondering if < 0 is correct or if this should be <= 0

See: https://github.com/postgres/postgres/blob/e09d763e25dccc40695bc824ddda9abea791d66f/src/backend/executor/execExpr.c#L915

Reply:
Fixed.